pkg/service/credential/service.go

```diff
@@ -256,6 +256,9 @@ func (s Service) signCredentialJWT(ctx context.Context, issuerKID string, cred c
 	if gotKey.Controller != cred.Issuer.(string) {
 		return nil, sdkutil.LoggingNewErrorf("key controller<%s> does not match credential issuer<%s> for key<%s>", gotKey.Controller, cred.Issuer, issuerKID)
 	}
+	if gotKey.Revoked {
```

Comment:
there are a few other places keys are used, we should check all those places (e.g. cred manifest signing, presentation request signing, schema signing, etc....)

Reply:
Agree, I [wanted to start small](https://github.com/TBD54566975/ssi-service/pull/546#issuecomment-1595522158) for a few reasons.

1. Align on the approach first, before I write a lot of code to be re-written later. Looks like @andresuribe87 [considers a different way](https://github.com/TBD54566975/ssi-service/pull/546#pullrequestreview-1488352322) to reflect revoked state. I will wait until you reach a consensus.

2. One big PR is harder to keep in sync with the mainline while it is being reviewed because there is a higher chance of getting a git conflict that will require extra work. I thought that `revoked` flag support could be rolled out incrementally in several smaller PRs. What do you think?

Reply:
I think approach this makes sense for now. We can tackle the API surface in a separate issue. I've created https://github.com/TBD54566975/ssi-service/issues/558 to track that. 

Thanks for tackling this!

Reply:
@andresuribe87 thanks for the update. I'm happy to contribute to transition to "state" in the future, if you settle on it.

I will push an update to address your other comments here.